low-level/files/opt/seagate/sspl/setup/resource_map/lr2/server.py

```diff
@@ -463,3 +465,18 @@ def get_nw_status(self, nw_interface, interface):
             nw_cable_conn_status = "UNKNOWN"
             logger.exception(self.log.svc_log(err))
         return nw_status, nw_cable_conn_status
+
+    def get_raid_info(self):
+        raids_data = []
+        for raid in RAIDs.get_configured_raids():
+            raid_data = self.get_health_template(raid.id, False)
+            health, description = raid.get_health()
+            devices = []
+            if health != "Fault":
```

Comment:
Why RAID devices info cant be queried if it has Fault ?

>        elif returncode == 4:
            return ("Fault", "There was an error while trying to get information about the array.")
        elif returncode == 2:
            return ("Failed", "The array has multiple failed devices such that it is unusable.")
            
           Going by this, if get_devices() cant be invoked, then in Failed case also it wont be possible ?

Reply:
Actually this is not needed. I have removed the check.
If there are no devices in output then this function will return []